examples/scripts/xpo.py

```diff
@@ -105,10 +95,6 @@ def prepare_dataset(row):
     with PartialState().local_main_process_first():
         dataset = dataset.map(prepare_dataset, num_proc=training_args.dataset_num_proc)
 
-    if args.max_samples is not None:
```

Comment:
FYI @kashif @qgallouedec @edbeeching we should not add this logic into the example scripts IMO - it's best solved by adding support for something like the dataset mixer we have in the handbook or H4 repo

Reply:
ah yes my bad!